src/coreclr/tools/Common/Internal/Runtime/EETypeBuilderHelpers.cs

```diff
@@ -91,6 +96,21 @@ public static ushort ComputeFlags(TypeDesc type)
             return flags;
         }
 
+        public static ushort ComputeFlagsEx(TypeDesc type)
+        {
+            ushort flagsEx = 0;
+
+            if (type is MetadataType mdType &&
+                            mdType.Module == mdType.Context.SystemModule &&
+                            (mdType.Name == "WeakReference" || mdType.Name == "WeakReference`1") &&
```

Comment:
`Name` is virtual call. The pattern matching code that I have shared was trying to avoid the redundant call of the virtual method.

Reply:
I thought it was a typo :-)

Reply:
Yeah, the flexibility of pattern matching syntax surprises me as well. I always have to check the IL to verify that it did what I have expected it to do.